sphere.cpp

```diff
@@ -0,0 +1,50 @@
+#include "sphere.h"
+
+Sphere::Sphere() {
+  for (Element element : ELEMENTLIST) {
+    insert({element, 0});
+  }
+}
+
+Sphere::Sphere(std::map<Element, int> gems) {
+  for (Element element : ELEMENTLIST) {
+    if (gems.find(element) != gems.end())
+      insert({element, gems.at(element)});
+    else
+      insert({element, 0});
+  }
+}
+
+Sphere::Sphere(std::vector<Element> gems) {
+  for (Element element : ELEMENTLIST) {
+    insert({element, 0});
+  }
```

Comment:
You should be able to use the other constructor instead of duplicating code:
```suggestion
Sphere::Sphere(std::vector<Element> gems) : Sphere() {
```